phpBB/phpbb/profilefields/type/type_interface.php

```diff
@@ -200,11 +208,11 @@ public function prepare_hidden_fields($step, $key, $action, &$field_data);
 	/**
 	* Allows assigning of additional template variables
 	*
-	* @param array	$template_vars	Template variables we are going to assign
+	* @param string	$action			Currently performed action (create|edit)
 	* @param array	$field_data		Array with data for this field
-	* @return null
+	* @return array	with the acp options for step 1
 	*/
-	public function display_options(&$template_vars, &$field_data);
+	public function display_options($action, &$field_data);
```

Comment:
Previous implementation of what?  With the proposed changes, the core code remains 100% functionally compatible, with no implications whatsoever, and no changes to the db either.  And as for the potential for some extensions to have used this...  the only one I have been able to find is my own Advanced Profile Field Types that triggered all these changes.  That does not mean that someone somewhere might have done something that might be affected, but up to what I have been able to find, nothing in phpbb forum is affected (again, as far as I have been able to see...).